scrapers/SmutPuppet.yml

```diff
@@ -0,0 +1,66 @@
+name: SmutPuppet
+sceneByURL:
+  - action: scrapeXPath
+    url: 
+      - blackandbig.com/update
+      - darksodomy.com/update
+      - dothewife.com/update
+      - dreamtranny.com/update
+      - genlez.com/update
+      - goldenslut.com/update
+      - grannyvsbbc.com/update
+      - jefssmodels.com/update
+      - milfsodomy.com/update
+      - smutmerchants.com/update
+      - suggabunny.com/update
+      - teenerotica.xxx/update
+    scraper: sceneScraper
+sceneByFragment:
+  action: scrapeXPath
+  queryURL: "{url}"
+  scraper: sceneScraper
+xPathScrapers:
+  sceneScraper:
+    scene:
+      Performers:
+        Name: //div[@class="model-rich"]/h4[@class="theme-color"]/a
+      Title: //div[@class="section-title"]/h4
+      Details:
+        selector: //p[@class="read-more"]/text()
+        postProcess:
+          - replace:
+              - regex: '^\s*:\s*'
+                with: ""
+      Date:
+        selector: //small[@class="updated-at"]/text()
+        postProcess:
+          - parseDate: Jan 2, 2006
+      Tags:
+        Name:
+          selector: //div[@class="model-categories"]/a/text()
+      Studio:
+        Name:
+          selector: //div[@class="block-logo"]/a/img/@alt
+      Image:
+        selector: //img[@class="video-banner"]/@src|//video/@poster
+        postProcess:
+          - replace:
+              - regex: (?:.+)(\/usermedia\/.+\.jpg)(?:.+)
+                with: $1
+              - regex: "^/usermedia/"
+                with: "https://smutpuppet.com/usermedia/"
+      Code:
+        selector: //div[@class="model-player"]/a/img/@src
+        postProcess:
+          - replace:
+              - regex: .+\/updates/(\d+)\/.+
+                with: $1
+      # Return the sanitized URL
+      URL:
+        selector: //div[@class="block-logo"]/a/img/@src|//div[@class="model-player"]/a/img/@src
+        concat: "|"
+        postProcess:
+          - replace:
+              - regex: \/static\/(\w+\....)\/.+\/(\d+)\/.+
+                with: https://$1/update/$2
```

Comment:
This works for every site except DoTheWife and Jeff'sModels

In addition the URL should include the trailing slash in order to be canonical; without this the site will redirect to the version with the slash anyway, so it's annoying when you rescrape a scene with a correct URL and get an additional nearly-identical URL in the list of URLs

The regex should be `https://$1/update/$2/`

Reply:
Fixed both issues.